### DISCUSSION TITLE
setting random location within a quadrant

### DISCUSSION THREAD

**Discussion by idoba2030** [jspsych/jsPsych#1316] [Q&A]
Hey everyone and thanks a lot for the help.

On each trial, I have a set of 4 or 8 squares that I want the subjects to see.
I want them to appear as 1 or 2 on each quadrant out of 3 possible locations in every quadrant.
How can I create a function to do it effectively? I wasn't able to find a solution...

Another issue is the different screen sizes that subjects may have. If I understand correctly, then I should create a <div> container in which the squares will appear with their relative (top,left) css attributes. Would that fix the problem or I need to somehow get the screen size of the subject?

Please let me know if you need me to upload a slice of the code I made this far.
Thanks a lot for your help!

**Comment by becky-gilbert** [comment]
Hi @idoba2030, I think this is a great case for using the new canvas-* plugins in jsPsych v6.2! You can define a canvas size either as a percentage of screen height/width or use fixed pixel values across all screen sizes. Then in your stimulus function, you can either draw the squares with canvas drawing functions or use images. You can then position them using the canvas' x-y coordinates, by calculating the x-y ranges for each quadrant and selecting random values within those ranges.

I hope this helps - feel free to post your code if you need more guidance!
